doc/protocol.md

```diff
@@ -757,6 +755,35 @@ The general open options are:
    When no "limit" is specified, all samples until the end of the
    archive are delivered.
 
+You specify the desired metrics as an array of objects, where each
+object describes one metric.  For example:
+
+    [ { name: "kernel.all.cpu.user",
+        units: "millisec",
+        derive: "rate"
+      },
+      ...
+    ]
+
+A metric description can contain the following fields:
+
+ * "name" (string): The name of the metric.  Available metrics depend
+   on the source.
+
+ * "units" (string, optional): The units that values for this metric
+   should be delivered in.  If the given metric can not be converted
+   into these units, the channel is closed.  The format of the string
+   depends on the source.
+
+ * "derive" (string, optional): Optional computation.  Possible values
+   are "delta" and "rate".  For "delta", the channel delivers the
+   difference between the current and the previous value for a metric.
+   For "rate", the channel delivers the change per millisecond of the
+   metric.
+
+   For both "delta" and "rate", the value for a metric will be "false"
+   if there is no previous value to do the computation with.
```

Comment:
Hmmm, i see "null" is used by the compression logic. So I guess "false" is a good choice. Although I can't help but wonder if zero ```0``` makes more sense. Anyway, this is something we can fine tune later, and not a blocker for this pull request.